packages/ckeditor5-find-and-replace/tests/ui/findandreplaceformview.js

```diff
@@ -67,333 +355,889 @@ describe( 'FindAndReplaceFormView', () => {
+			} );
+
+			it( 'should adjust the right padding of the find input depending on the presence of the counter', () => {
+				editor.setData( `<p>${ new Array( 20 ).join( 'A' ) }</p>` );
+				openDropdown();
+
+				findInput.fieldView.value = 'A';
+
+				expect( findInput.fieldView.element.style.paddingRight ).to.equal( '' );
+
+				findButton.fire( 'execute' );
+
+				expect( matchCounterElement.textContent ).to.equal( '1 of 19' );
+				expect( findInput.fieldView.element.style.paddingRight ).to.match( /^calc/ );
 
-		it( 'returns the #replaceInputView DOM value', () => {
-			viewValue = view.replaceInputView.fieldView.element.value;
-			viewValue = 'foo';
+				findInput.fieldView.value = 'AA';
+				findInput.fieldView.fire( 'input' );
 
-			expect( viewValue ).to.equal( 'foo' );
+				expect( findInput.fieldView.element.style.paddingRight ).to.equal( '' );
+
+				findButton.fire( 'execute' );
+
+				expect( matchCounterElement.textContent ).to.equal( '1 of 9' );
+				expect( findInput.fieldView.element.style.paddingRight ).to.match( /^calc/ );
+			} );
 		} );
 
-		it( 'sets the #findInputView DOM value', () => {
-			viewValue = view.findInputView.fieldView.element.value;
-			viewValue = 'bar';
-			viewValue = 'foo';
+		describe( 'using the replace UI', () => {
+			it( 'should bind "replace" button #isEnabled to the "replace" command', () => {
+				editor.setData( '<p>AAA</p>' );
+				openDropdown();
+
+				findInput.fieldView.value = 'A';
+				findButton.fire( 'execute' );
+
+				const command = editor.commands.get( 'replace' );
+
+				command.isEnabled = false;
+				expect( replaceButton.isEnabled ).to.be.false;
+
+				command.isEnabled = true;
+				expect( replaceButton.isEnabled ).to.be.true;
+			} );
+
+			it( 'should bind "replace all" button #isEnabled to the "replaceAll" command', () => {
+				editor.setData( '<p>AAA</p>' );
+				openDropdown();
+
+				findInput.fieldView.value = 'A';
+				findButton.fire( 'execute' );
+
+				const command = editor.commands.get( 'replaceAll' );
+
+				command.isEnabled = false;
+				expect( replaceAllButton.isEnabled ).to.be.false;
+
+				command.isEnabled = true;
+				expect( replaceAllButton.isEnabled ).to.be.true;
+			} );
+
+			it( 'should bind replace input #isEnabled to the "replace" command', () => {
+				editor.setData( '<p>AAA</p>' );
+				openDropdown();
+
+				findInput.fieldView.value = 'A';
+				findButton.fire( 'execute' );
+
+				const command = editor.commands.get( 'replace' );
+
+				command.isEnabled = false;
+				expect( replaceInput.isEnabled ).to.be.false;
+
+				command.isEnabled = true;
+				expect( replaceInput.isEnabled ).to.be.true;
+			} );
+
+			it( 'should display a tip when the replace field is disabled but not focused', () => {
+				openDropdown();
+
+				expect( replaceInput.isEnabled ).to.be.false;
+				expect( replaceInput.infoText ).to.equal( '' );
+			} );
+
+			it( 'should display a tip when the replace field is disabled and focused', () => {
```

Comment:
This test fails when the Chrome window, that executes the tests, is not focused. @oleq, AFAIK, you created a workaround some time ago for a similar problem, right?